### PR TITLE
Added CHIP shell commands to set pincode and discriminator

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
Currently CHIP shell config commands allow only to get config parameters values. Allowing to set pincode and discriminator
values used in commissioning would help in testing and enable radomizing those values to not use always the same ones.

#### Change overview
* Added commands to set pincode and discriminator values.
* Added returning errors in case number of command attributes is different than expected.

#### Testing
Small change, so only tested manually that calling set commands for pincode and discriminator works fine, as presented below:
```
$ matter config pincode
20202021
$ matter config pincode 98766789
Setup pin code set to: 98766789
$ matter config pincode
98766789
$ matter config discriminator
f00
$ matter config discriminator 3841
Setup discriminator set to: 3841
$ matter config discriminator
f01
```

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: src/lib/shell/commands/Config.cpp
Comment (tcarmelveilleux) [inline src/lib/shell/commands/Config.cpp:103]: The device should not need to know the "PIN code" (i.e. commissioning passcode), just the w0/L of the verifier. We use PIN today because we did not design an easy way to have the onboarding payload separate from running binary, but it is problematic since it forces embedded target to compute the verifier itself, which in turn forces us to use an iteration count that is too small for PBKDF.

Obviously it would be be disruptive to remove the passcode in the short term, but adding more new commands that enshrine usage that doesn't match to spec intent make the transition to the final approach more problematic

Comment (kkasperczyk-no) [reply src/lib/shell/commands/Config.cpp:103]: Ok, I got your point, then would you be ok with adding a command only to set discriminator? That should be enough to clearly distinguish devices that we want to commission, but would not introduce further deviations from spec.

**Review by woody-apple — APPROVED** [review]

File: src/lib/shell/commands/Config.cpp
Comment (tcarmelveilleux) [reply src/lib/shell/commands/Config.cpp:103]: Yes, definitely you can keep the discriminator setter and I wouldn't be against having a short term passcode setter until we sort it all out, but this is an area where we have to start to move towards more realistic usage and APIs for accessing the data. Will open an issue for discussion since my Slack thread did not get traction

**Review by tcarmelveilleux — APPROVED** [review]
Approve assuming this is not forever, since devices do not need in general to know their own passcode unless they specifically need to render a QR or present it on display

File: src/lib/shell/commands/Config.cpp
Comment (kkasperczyk-no) [reply src/lib/shell/commands/Config.cpp:103]: So what is the conclusion, should I remove setter for passcode now, or we want to leave it and create issue, as a reminder to do that? Sorry, but after your approval I a little bit got lost and I don't know if I can to resolve conversation?

**Comment by kkasperczyk-no** [comment]
@andy31415 @msandstedt @jepenven-silabs @hawk248 @chrisdecenzo could somebody review? It needs one more approve

**Review by msandstedt — APPROVED** [review]

**Pull request merged**
